src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    private static class Node<K, V> {
+        private final K key;
+        private Node<K, V> next;
+        private V value;
+
+        private Node(V value, K key, Node<K, V> next) {
+            this.value = value;
+            this.key = key;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null || threshold == size) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(value, key, null);
+        if (table[getIndex(key)] == null) {
+            table[getIndex(key)] = newNode;
+            size++;
+        } else {
+            Node<K, V> currentNode = table[getIndex(key)];
+            while (currentNode.next != null && !Objects.equals(currentNode.key, key)) {
+                currentNode = currentNode.next;
+            }
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+            } else {
+                currentNode.next = newNode;
+                size++;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> currentNode = findCurrentNode(key);
+        return currentNode != null ? currentNode.value : null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(Object key) {
+        if (key == null) {
+            return 0;
+        }
+        return Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        if (table == null) {
+            table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
```

Comment:
May be better initialize table in constructor?
If you do it, thinking about null check.